services/cohort/index.ts

```diff
@@ -0,0 +1,144 @@
+import type { CohortDTO } from './types';
+import { cohortDTOToCohort } from './utils';
+import {
+  getGraphQLURL,
+  gqlQueryOptions,
+  getItemsURL,
+  apiRestOptions,
+} from '@/services/utils';
+import { useUserStore } from '@/store/user/store';
+import type { Query } from '@/store/types';
+import type { Cohort, Member } from '@/types/cohort';
+import type { Course } from '@/types/course';
+import { courseDTOToCourse } from '@/services/course/utils';
+import { CourseDTO } from '@/services/course/types';
+
+const COHORTS_URL = `${getItemsURL()}/cohorts`;
+
+export interface CohortsAndCourseDTO {
+  cohorts: CohortDTO[];
+  courses: CourseDTO[];
+}
+
+export const getCohortsByCourse = async (
+  courseId: Course['id'],
+  language = 'en-US'
+) => {
+  const query = /* GraphQL */ `
+    query GetCohortsByCourse($courseId: String!, $language: String!) {
+      courses(filter: { id: { _eq: $courseId } }) {
+        slug
+        id
+        translations(filter: { languages_code: { code: { _eq: $language } } }) {
+          title
+        }
+        units(limit: 1) {
+          units_id {
+            slug
+          }
+        }
+      }
+      cohorts(filter: { courses: { courses_id: { _eq: $courseId } } }) {
+        id
+        name
+        invitations
+        auto_enrollment
+        enrollment_code
+        members {
+          id
+          directus_users_id {
+            email
+          }
+        }
+      }
+    }
+  `;
+
+  const queryVariables = { courseId, language };
+
+  const res = await $fetch<Query<CohortsAndCourseDTO>>(
+    getGraphQLURL(),
+    gqlQueryOptions(query, queryVariables)
+  );
+
+  const cohorts = res.data.cohorts.map(cohortDTOToCohort);
+  const course = courseDTOToCourse(res.data.courses[0]);
+
+  return { course, cohorts };
+};
+
+export const createCohort = async (
+  courseId: Course['id'],
+  name: Cohort['name'],
+  invitations: Cohort['invitations'],
+  autoEnrollment: Cohort['autoEnrollment']
+) => {
+  const userStore = useUserStore();
+  const { currentUser } = storeToRefs(userStore);
+
+  const body = {
+    name,
+    invitations,
+    auto_enrollment: autoEnrollment,
+    courses: {
+      create: [
+        {
+          courses_id: {
+            id: courseId,
+          },
+        },
+      ],
+    },
+    admins: {
+      create: [
+        {
+          directus_users_id: {
+            id: currentUser.value?.id,
```

Comment:
```suggestion
            id: currentUser.value.id,
```

@Tansito We cannot allow this. Maybe we need another check or a `!`.

Reply:
Yeah, totally. I added a `!` because initially this function can only be called authenticated but we could add an `if` before just to ensure us.

Reply:
Done [c956ae6](https://github.com/Qiskit/saiba/pull/336/commits/c956ae664a904eafb1d08cad6358fcebf34e700b)